sdk/go/common/apitype/plan.go

```diff
@@ -0,0 +1,89 @@
+package apitype
+
+import (
+	"encoding/json"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+)
+
+// PlanDiffV1 is the serializable version of a plan diff.
+type PlanDiffV1 struct {
+	// the resource properties that will be added.
+	Adds map[string]interface{} `json:"adds,omitempty"`
+	// the resource properties that will be deleted.
+	Deletes []string `json:"deletes,omitempty"`
+	// the resource properties that will be updated.
+	Updates map[string]interface{} `json:"updates,omitempty"`
+}
+
+// GoalV1 is the serializable version of a resource goal state.
+type GoalV1 struct {
+	// the type of resource.
+	Type tokens.Type `json:"type"`
+	// the name for the resource's URN.
+	Name tokens.QName `json:"name"`
+	// true if this resource is custom, managed by a plugin.
+	Custom bool `json:"custom"`
+	// the resource properties that will be changed.
+	CheckedInputs map[string]interface{} `json:"checkedInputs,omitempty"`
+	// the resource properties that will be changed.
+	InputDiff PlanDiffV1 `json:"inputDiff,omitempty"`
+	// the resource outputs that will be changed.
+	OutputDiff *PlanDiffV1 `json:"outputDiff,omitempty"`
```

Comment:
Is the *PlanDiffV1 intentional vs PlanDiffV1 in InputDiff? Extra meaning for nil?

Reply:
Huh yeh I think at one point nil meant something special. I think its just the same as empty now I'll try and clean up.